### PR TITLE
Make sSetupDLs a Gfx[][6]

### PR DESCRIPTION
**Pull request by petrie911**
This array makes more sense as a 2D array, although one function has to do some fake stuff to match. It's possible the game didn't have it as a 2D array originally, though I can't imagine why that would be.

### CODE REVIEW

File: src/code/z_rcp.c [hunk 1]
```diff
@@ -1,737 +1,881 @@
...
+        gsDPSetCombineLERP(PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, PRIMITIVE, 0, TEXEL0, 0, 0, 0, 0, COMBINED, 0,
+                           0, 0, COMBINED),
+        gsDPSetOtherMode(G_AD_NOISE | G_CD_NOISE | G_CK_NONE | G_TC_FILT | G_TF_BILERP | G_TT_NONE | G_TL_TILE |
+                             G_TD_CLAMP | G_TP_PERSP | G_CYC_2CYCLE | G_PM_NPRIMITIVE,
+                         G_AC_NONE | G_ZS_PIXEL | G_RM_FOG_SHADE_A | G_RM_ZB_CLD_SURF2),
+        gsSPLoadGeometryMode(G_ZBUFFER | G_SHADE | G_CULL_BACK | G_FOG | G_SHADING_SMOOTH),
+        gsSPEndDisplayList(),
+    },
+
+    {
```
Comment (Dragorn421) [inline src/code/z_rcp.c:17]: I would get rid of the empty lines between the entries

**Review by AngheloAlf — APPROVED** [review]

File: src/code/z_rcp.c [hunk 2]
```diff
@@ -821,7 +895,10 @@ Gfx* Gfx_SetFog2(Gfx* gfx, s32 r, s32 g, s32 b, s32 a, s32 n, s32 f) {
 }
 
 Gfx* Gfx_CallSetupDLImpl(Gfx* gfx, u32 i) {
-    gSPDisplayList(gfx++, &sSetupDL[6 * i]);
+    s32 dListIndex = 6 * i;
+
+    gSPDisplayList(gfx++, &((Gfx*)sSetupDL)[dListIndex]);
```
Comment (fig02) [inline src/code/z_rcp.c:900]: i still feel like doing stuff like this is forcing it to be something that its not. but i guess we've always taken our liberties with changing the structure of something to make it better if it doesnt effect matching. 
just still dont feel 100% about it in this case

Comment (petrie911) [reply src/code/z_rcp.c:900]: I might be forcing it to be something it wasn't, but I'm not forcing it to be something it's not. It *is* a n x 6 array of Gfx commands. That Nintendo chose a strange way to handle it shouldn't make a difference.

Comment (fig02) [reply src/code/z_rcp.c:900]: fair enough, we're looking to do the same thing with vrbox anyway. Still dont feel 100% about it, but will go with the majority here.
Which btw if you want to make the changes to the skybox palletes, is prob the only thing holding that up

**Review by fig02 — APPROVED** [review]

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**